### PR TITLE
chore(dev): Wrap `vector-config` in `vector-lib` as `configurable`

### PR DESCRIPTION
**Pull request by bruceg**
There already exists a `vector_lib::config` export so I figured adding `vector-config` to that module would make things too confusing. Since most of `vector-config` has to do with the `configurable` terminology, that seemed a natural fit for this export.

### CODE REVIEW

File: lib/vector-lib/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,10 @@ edition = "2021"
 publish = false
 
 [dependencies]
-vector-common = { path = "../vector-common", default-features = false }
+vector-common = { path = "../vector-common" }
+vector-config = { path = "../vector-config" }
+vector-config-common = { path = "../vector-config-common" }
+vector-config-macros = { path = "../vector-config-macros" }
```
Comment (tobz) [inline lib/vector-lib/Cargo.toml:12]: I wonder if, actually, with your changes.... needing to import `vector-config-common` and `vector-config-macros` directly can now be dropped? 🤔

Comment (tobz) [reply lib/vector-lib/Cargo.toml:12]: If so, it would also let you significantly shrink/clean up the `pub mod configurable` block in `vector-lib`.

**Review by tobz — APPROVED** [review]
Overall, this looks fine to me if it passes CI.

Left one thought about maybe being able to clean up some of the dependencies/re-exports.

File: lib/vector-lib/Cargo.toml (hunk 1 above)
Comment (bruceg) [reply lib/vector-lib/Cargo.toml]: Great catch! Fixed in 7983942e1

**Pull request merged**